snuba/web/rpc/v1/resolvers/common/aggregation.py

```diff
@@ -463,182 +463,147 @@ def get_extrapolated_function(
     return function_map_sample_weighted.get(aggregation.aggregate)
 
 
-def get_confidence_interval_column(
+def _get_ci_count(
     aggregation: AttributeAggregation | AttributeConditionalAggregation,
-) -> Expression | None:
+    alias: str | None = None,
+    z_value: float = Z_VALUE_P95,
+) -> Expression:
     """
-    Returns the expression for calculating the upper confidence limit for a given aggregation. If the aggregation cannot be extrapolated, returns None.
-    Calculations are based on https://github.com/getsentry/extrapolation-math/blob/main/2024-10-04%20Confidence%20-%20Final%20Approach.ipynb
-    Note that in the above notebook, the formulas are based on the sampling rate, while we perform calculations based on the sampling weight (1 / sampling rate).
+    confidence interval = Z \cdot \sqrt{\sum_{i=1}^n w_i^2 - w_i}
+
+          ┌───────────┐
+          │ ₙ
+       ╲  │ ⎲   2
+    Z ⋅ ╲ │ ⎳  wᵢ - wᵢ
+         ╲│ⁱ⁼¹
+
+    where w_i is the sampling weight for the i-th event and n is the number of events.
     """
+
     field = attribute_key_to_expression(aggregation.key)
-    alias = get_attribute_confidence_interval_alias(aggregation)
+    condition_in_aggregation = _get_condition_in_aggregation(aggregation)
     alias_dict = {"alias": alias} if alias else {}
 
+    variance = f.sumIf(
+        f.minus(
+            f.multiply(sampling_weight_column, sampling_weight_column),
+            sampling_weight_column,
+        ),
+        and_cond(
+            get_field_existence_expression(field),
+            condition_in_aggregation,
+        ),
+    )
+
+    return f.multiply(z_value, f.sqrt(variance), **alias_dict)
+
+
+def _get_ci_sum(
+    aggregation: AttributeAggregation | AttributeConditionalAggregation,
+    alias: str | None = None,
+    z_value: float = Z_VALUE_P95,
+) -> Expression:
+    """
+    confidence interval = Z \cdot \sqrt{\sum_{i=1}^n x_i^2 \cdot (w_i^2 - w_i)}
+
+          ┌─────────────────┐
+          │ ₙ
+       ╲  │ ⎲   2    2
+    Z ⋅ ╲ │ ⎳  xᵢ ⋅(wᵢ - wᵢ)
+         ╲│ⁱ⁼¹
+    """
```

Comment:
In the PR description, you said this was using the [Horvitz-Thompson estimator](https://en.wikipedia.org/wiki/Horvitz%E2%80%93Thompson_estimator) method. I don't think I understand how this formula maps to the method

Reply:
Ah yes, I'll update that description and make it less cryptic 👍

Reply:
To answer your question, page 9 / formula 9 of [the paper](https://www.stat.cmu.edu/~brian/905-2008/papers/Horvitz-Thompson-1952-jasa.pdf) is where this comes from. There's a great [blog post](https://blog.cloudflare.com/how-we-make-sense-of-too-much-data/#extracting-value-from-downsampled-data) from cloudflare that elaborates on the same approach.

I've reconsidered and don't think we should go beyond stating which estimator and variance method we use. However, we can put more reasoning in our [math repo](https://github.com/getsentry/extrapolation-math/tree/main) (I've not updated that yet) and then link it, WDYT?